### PR TITLE
Add Featureset class extending xr.Dataset

### PR DESCRIPTION
**Pull request by bnaul**
- Implements imputation (constant value + mean/median/mode)
- Overloads indexing to simplify extracting features for specific time series

### CODE REVIEW

File: cesium/tests/test_featureset.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+DATA_PATH = pjoin(os.path.dirname(__file__), "data")
+
+
+def test_repr():
+    """Testing Featureset printing."""
+    fset = Featureset(xr.open_dataset(pjoin(DATA_PATH, "test_featureset.nc")))
+    repr(fset)
+
+
+def test_impute():
```
Comment (acrellin) [inline cesium/tests/test_featureset.py]: Shouldn't we be testing the other impute strategies? At least to catch any API/function signature changes on the part of `sklearn`.

Comment (bnaul) [inline cesium/tests/test_featureset.py]: Good point, added some addition tests :+1:

**Comment by stefanv** [comment]
Eek, merge conflicts!

**Comment by bnaul** [comment]
Don't worry about it, this can wait. Just adds some more convenient syntax
options.

**Comment by stefanv** [comment]
Should we get this merged and cleared out of the pipeline?

**Comment by bnaul** [comment]
I think this is good to go now; the front end tests will break temporarily because this changes the signature of one of the test fixture functions that's also used by the front end, but after merging I can do a release and push a fix to `cesium_web`.

**Comment by stefanv** [comment]
Other than the cryptic comment on line 10 (which should probably go or get expanded), this looks great.

**Comment by acrellin** [comment]
:+1:

**Pull request merged**